crates/fullnode/src/runner.rs

```diff
@@ -421,63 +420,45 @@ where
             bail!("Previous hash mismatch at height: {}", l2_height);
         }
 
-        let mut data_to_commit = SlotCommit::new(current_l1_block.clone());
+        // let mut data_to_commit = SlotCommit::new(current_l1_block.clone());
 
         let pre_state = self
             .storage_manager
             .create_storage_on_l2_height(l2_height)?;
 
-        let slot_result = self.stf.apply_soft_confirmation(
-            self.fork_manager.active_fork().spec_id,
-            self.sequencer_pub_key.as_slice(),
-            // TODO(https://github.com/Sovereign-Labs/sovereign-sdk/issues/1247): incorrect pre-state root in case of re-org
-            &self.state_root,
-            pre_state,
-            Default::default(),
-            current_l1_block.header(),
-            &current_l1_block.validity_condition(),
-            &mut soft_confirmation.clone().into(),
-        );
+        let soft_confirmation_result = self
+            .stf
+            .apply_soft_confirmation(
+                self.fork_manager.active_fork().spec_id,
+                self.sequencer_pub_key.as_slice(),
+                // TODO(https://github.com/Sovereign-Labs/sovereign-sdk/issues/1247): incorrect pre-state root in case of re-org
+                &self.state_root,
+                pre_state,
+                Default::default(),
```

Comment:
what is that default arg? Maybe we can get rid of it?

Reply:
it's witness, node doesn't need it but ZK guest does.